MandatoryTask/MandatoryTask/CandyGiftBasket.java

```diff
@@ -0,0 +1,110 @@
+package MandatoryTask;
+
+import java.util.*;
+
+public class CandyGiftBasket {
+    private List<Candy> candies;
+    private final int MAX_WEIGHT_IN_GRAMS = 500;
+    private final int MAX_CALORIFIC = 2500;
+    private final int MAX_CANDY_COUNT = 10;
+
+    public CandyGiftBasket() {
+        candies = new ArrayList<>();
+    }
+
+    public void addCandy(Candy candy) throws GiftBasketOvercalorificException, GiftBasketOverweightException, MaxCandyExceededException, NullPointerException {
+        if (candy.getName() != null && candy.getWeightOfOneCandyGram() > 0 && candy.getCalorificPerGram() > 0) {
+            if (verifyIsNotOvercalorific(candy) && verifyIsNotOverweight(candy) && verifyIsNotExceedMaxNumberOfCandies(candy)) {
+                candies.add(candy);
+            } else if (!verifyIsNotOvercalorific(candy)) {
+                throw new GiftBasketOvercalorificException("The GiftBasket calorific exceeds 2500 calories");
+            } else if (!verifyIsNotOverweight(candy)) {
+                throw new GiftBasketOverweightException("The GiftBasket weight exceeds 500g");
+            } else if (!verifyIsNotExceedMaxNumberOfCandies(candy)) {
+                throw new MaxCandyExceededException("Candy count exceeds 10 in the GiftBasket");
+            }
+        } else {
+            throw new NullPointerException("Some candy values are empty or null. Please, re-check it");
+        }
+    }
```

Comment:
і тоді метод можна скоротити до такого
```suggestion
    public void addCandy(Candy candy) {
        try {
            verifyIsNotOverweight(candy);
            verifyIsNotOvercalorific(candy);
            verifyIsNotOvercalorific(candy);
            candies.add(candy);
        } catch (SomeGeneralExxception e) {
            System.err.println(e.getMessage());
        }
    }
```